src/app-layer-ssl.c

```diff
@@ -691,6 +694,10 @@ static inline int TLSDecodeHSHelloVersion(SSLState *ssl_state,
     uint16_t version = (uint16_t)(*input << 8) | *(input + 1);
     ssl_state->curr_connp->version = version;
 
+    if (SC_ATOMIC_GET(ssl_config.enable_ja4) && ssl_state->curr_connp->ja4 != NULL) {
```

Comment:
First check is unnecessary if we have the second one, right ?

Reply:
Yes, because `ssl_state->curr_connp->ja4` is only set to a non-`NULL` value if `SC_ATOMIC_GET(ssl_config.enable_ja4)` is true. Should I remove that in another iteration?

Reply:
I think testing local `ssl_state->curr_connp->ja4` is cheaper than testing atomic global...

Reply:
> And we do not test if it is from client here cf `SSL_AL_FLAG_STATE_CLIENT_HELLO`

This is also already checked when setting the `ssl_state->curr_connp->ja4` value to not-NULL in https://github.com/OISF/suricata/blob/f7389fa100f30b917299621df3ea8d9403b46ac9/src/app-layer-ssl.c#L1531

```C
    /* Ensure that we have a JA4 state defined by now if we have JA4 enabled,
       we are in a client hello and we don't have such a state yet (to avoid
       leaking memory in case this function is entered more than once). */
    if (SC_ATOMIC_GET(ssl_config.enable_ja4) &&
            ssl_state->current_flags & SSL_AL_FLAG_STATE_CLIENT_HELLO &&
            ssl_state->curr_connp->ja4 == NULL) {
        ssl_state->curr_connp->ja4 = SCJA4New();
    }
```
So if `ssl_state->curr_connp->ja4` is not `NULL` then `SSL_AL_FLAG_STATE_CLIENT_HELLO` is already checked

Reply:
I'll simplify these checks everywhere.

Reply:
> So if ssl_state->curr_connp->ja4 is not NULL then SSL_AL_FLAG_STATE_CLIENT_HELLO is already checked

On creation, but server hello has not `SSL_AL_FLAG_STATE_CLIENT_HELLO` but ssl_state->curr_connp->ja4 is not NULL

Reply:
Right, the same function is called multiple times for both handshake sides. Keeping the `SSL_AL_FLAG_STATE_CLIENT_HELLO` check